presto-manager-agent/src/main/java/com/teradata/prestomanager/agent/api/PackageAPI.java

```diff
@@ -36,87 +35,94 @@
+            @ApiResponse(code = 400, message = "Invalid url"),
+            @ApiResponse(code = 409, message = "Presto is running")
     })
-    public synchronized Response upgrade(@NotNull @ApiParam("Url to fetch package") String locationToFetchPackage,
+    public synchronized Response upgrade(@ApiParam("Url to fetch package") String locationToFetchPackage,
             @QueryParam("checkDependencies") @DefaultValue("true") @ApiParam("If false, disables dependency checking") boolean checkDependencies)
     {
         LOGGER.debug("POST /package  url: {} ; checkDependencies: {}", locationToFetchPackage, checkDependencies);
+        if ("".equals(locationToFetchPackage)) {
+            LOGGER.error("Url is empty or null");
+            return Response.status(BAD_REQUEST).entity("Expected URL in the body").build();
+        }
         try {
-            PackageType packageType = PackageType.valueOf(getFileExtension(locationToFetchPackage).toUpperCase());
-            if (installedPackageType != packageType) {
-                LOGGER.error("Package type mismatch. Presto installed using {} , but attempting to upgrade using {}", installedPackageType, packageType);
-                return Response.status(BAD_REQUEST).entity(text("Provided package format doesn't match the installed package format")).build();
+            if (isRunning()) {
+                return Response.status(CONFLICT).entity("Presto is running").build();
             }
+            URL url = new URL(locationToFetchPackage);
+            new Thread(new PrestoAsynchronousCommand(new PrestoUpgrader(PackageType.RPM, url, checkDependencies))).start();
         }
-        catch (IllegalArgumentException e) {
-            LOGGER.error("Invalid package: {}. Package must be .tar.gz or .rpm", locationToFetchPackage, e);
-            return Response.status(BAD_REQUEST).entity(text("Invalid package type for presto-server. Package must be .tar.gz or .rpm")).build();
-        }
-        try {
-            final URL url = new URL(locationToFetchPackage);
-            new Thread(new PrestoAsynchronousCommand(new PrestoUpgrader(installedPackageType, url, checkDependencies))).start();
+        catch (PrestoManagerException e) {
+            LOGGER.error("Failed to ascertain whether presto is running", e);
+            return Response.status(INTERNAL_SERVER_ERROR).entity("Failed to ascertain whether presto is running").build();
         }
         catch (MalformedURLException e) {
             LOGGER.error("Invalid url: {}", locationToFetchPackage, e);
-            return Response.status(BAD_REQUEST).entity(text("Invalid url")).build();
+            return Response.status(BAD_REQUEST).entity("Invalid url").build();
         }
-        return Response.status(ACCEPTED).entity(text("Check status to make sure that presto is upgraded")).build();
+        return Response.status(ACCEPTED).entity("Presto is being upgraded. Please check back later using status.").build();
```

Comment:
Either don't specify how to check back, or be more explicit.
`"Presto is being upgraded. Check installation status at '../presto/status'"`